server/src/endpoints/qs/push_notification_provider.rs

```diff
@@ -111,77 +239,143 @@ impl ProductionPushNotificationProvider {
 
         Ok(token)
     }
+
+    async fn push_google(&self, push_token: PushToken) -> Result<(), PushNotificationError> {
+        // If we don't have an FCM state, we can't send push notifications
+        let Some(fcm_state) = &self.fcm_state else {
+            return Ok(());
+        };
+
+        let service_account = &fcm_state.service_account;
+
+        let bearer_token = self
+            .issue_fcm_token()
```

Comment:
This will delay a notification each hour when the token expires. We should pro-actively spawn a task to refresh the token before expiration. I would leave it outside of this PR as technical debt though.

Reply:
Yes, I filed #237.